pystac_client/item_search.py

```diff
@@ -669,22 +622,61 @@ def items_as_dicts(self) -> Iterator[Dict[str, Any]]:
                 if self._max_items and nitems >= self._max_items:
                     return
 
+    # ------------------------------------------------------------------------
+    # By Page
+    def pages(self) -> Iterator[ItemCollection]:
+        """Iterator that yields ItemCollection objects.  Each ItemCollection is
+        a page of results from the search.
+
+        Yields:
+            ItemCollection : a group of Items matching the search criteria within an
+            ItemCollection
+        """
+        if isinstance(self._stac_io, StacApiIO):
+            for page in self.pages_as_dicts():
+                yield ItemCollection.from_dict(
+                    page, preserve_dict=False, root=self.client
+                )
+
+    def pages_as_dicts(self) -> Iterator[Dict[str, Any]]:
+        if isinstance(self._stac_io, StacApiIO):
+            for page in self._stac_io.get_pages(
+                self.url, self.method, self.get_parameters()
+            ):
+                yield page
+
+    # ------------------------------------------------------------------------
+    # Everything
+
     @lru_cache(1)
-    def get_all_items_as_dict(self) -> Dict[str, Any]:
-        """DEPRECATED. Use :meth:`get_items` or :meth:`get_item_collections` instead.
-            Convenience method that gets all items from all pages, up to
-            the number provided by the max_items parameter, and returns an array of
-            dictionaries.
+    def item_collection(self) -> ItemCollection:
+        """
+        Get the matching items as a :ref:`pystac.ItemCollection`.
 
         Return:
-            Dict : A GeoJSON FeatureCollection
+            item_collection: ItemCollection
         """
-        warnings.warn(
-            "get_all_items_as_dict is deprecated, use get_items or"
-            " get_item_collections instead",
-            DeprecationWarning,
-            stacklevel=2,
+        # Bypass the cache here, so that we can pass __preserve_dict__
```

Comment:
FYI: bypassing the `lru_cache` here! I *think* this also affected `main` but I haven't checked.